src/core/runQuery.ts

```diff
@@ -32,7 +32,19 @@ export interface QueryOptions {
  formatResponse?: Function;
 }
 
+// Make the global Promise constructor Fiber-aware.
+import { makeCompatible } from 'meteor-promise';
+import Fiber = require('fibers');
+makeCompatible(Promise, Fiber);
+
+const resolvedPromise = Promise.resolve();
+
 function runQuery(options: QueryOptions): Promise<GraphQLResult> {
+    // Fiber-aware Promises run their .then callbacks in Fibers.
+    return resolvedPromise.then(() => doRunQuery(options));
```

Comment:
Actually I think this may be the key to making everything work in Meteor, since the `Promise` that Meteor uses already knows how to run its callbacks in `Fiber`s. In other words, there's not actually any additional work to do on the https://github.com/apollostack/meteor-integration side!